doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,743 @@
+# Glossary
+
+This page is a glossary for various terms used throughout the SymPy
+documentation. This glossary is primarily for terms that are specific to
+SymPy. For more general Python terms, refer to the [Python
+glossary](https://docs.python.org/3/glossary.html). Mathematical terms are
+only included here if they have a specific meaning in SymPy. For general
+mathematical definitions, refer to other sources such as
+[Wikipedia](https://www.wikipedia.org/) or
+[MathWorld](https://mathworld.wolfram.com/), as well as the references in the
+documentation for the specific SymPy functions.
+
+<!-- Please follow the following style for the glossary:
+
+- Capitalize the names, unless they refer to code.
+- Each definition should be short (no more than a paragraph).
+- Keep the terms in alphabetical order.
+- Italicize the first occurrence of the term in the definition.
+- Use cross-references to refer to full documentation for functions, etc.
+- Cross-reference the first instance of other terms in definitions (see
+  below).
+- Classes and functions should only be listed here if they are core concepts
+  for understanding how to use SymPy. Classes and functions themselves should
+  be documented in their respective docstrings in the modules reference.
+
+-->
+
+<!-- To cross reference a glossary item elsewhere in the documentation use
+{term}`termname` in Markdown or :term:`termname` in RST. Use
+{term}`Text <termname>` or :term:`text <termname>` to make a link with
+different text. -->
+
+```{glossary}
+
+Antiderivative
+
+    An *antiderivative* of a function $f(x)$ with respect to $x$ is an
+    function $F(x)$ such that $\frac{d}{dx}F(x) = f(x).$ It is also sometimes
+    called an "indefinite integral" of $f(x)$, and written as $\int
+    f(x)\,dx.$ Antiderivatives in SymPy can be computed with
+    {func}`~.integrate`. Any given function has an infinite family of
+    antiderivatives, typically obtained by adding an arbitrary constant, but
+    {func}`~.integrate` only returns one
+    such antiderivative, which is sometimes informally referred to as "the
+    antiderivative" of $f(x)$. Note some sources call this the "primitive" of
+    $f(x)$, but this terminology is not used in SymPy because it is not as
+    universally used as "antiderivative", and because "primitive" has other
+    meanings in mathematics.
+
+`args`
+
+    The *`args`* property of a SymPy {term}`expression` `expr` is a tuple of
+    the top-level {term}`subexpressions <subexpression>` used to create
+    `expr`. They are the arguments to the class used to create the expression.
+    The args of any expression can be obtained by the `.args` attribute. For
+    example, `(1 + x*y).args` is `(1, x*y)`, because it equals `Add(1, x*y)`.
+    The `args` together with {term}`func` completely define an expression. It
+    is always possible to walk the {term}`expression tree` and extract any
```

Comment:
Recommend changing "_walk_ the expression tree" with "_traverse_" to

- make the analogy with traversing the DOM (document object model), which is a common phrase in web development
- be more inclusive (less ableist)

Reply:
Would you be happy with language that uses both, like "walk/traverse the expression tree"? Grepping, I see both uses in various places, including a function called `walk` that does just this.

The DOM is also a tree, so similar language will be used when talking about it. We can also add a glossary entry for walk/traverse. Another tree structure people may be familiar with is a file system. I think it depends on people's backgrounds which analogies they may be familiar with.

I linked to the Wikipedia article https://en.wikipedia.org/wiki/Tree_(data_structure) in the "expression tree" definition which lists the DOM as an example, but I'm happy to link to other, better references as well.

Reply:
While I think someone from the disability community would be better to answer that question, I appreciate that there are deeply embedded terms such as `walk` and that changing them all would be difficult for maintainers, and confusing for users. So yes, I think using both terms is a good idea at present.

That Wikipedia link seems good and sufficient.